components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java

```diff
@@ -92,6 +93,297 @@
+     *
+     * @param resourceConfigsJSON swaggerJSON
+     * @return URI Templates
+     * @throws APIManagementException
+     */
+    @Override
+    public Set<URITemplate> injectOtherResourceScopesToDefaultScheme(String resourceConfigsJSON) throws APIManagementException {
+        Swagger swagger = getSwagger(resourceConfigsJSON);
+        Set<URITemplate> urlTemplates = new LinkedHashSet<>();
+        Set<Scope> scopes = injectOtherScopesToDefaultScheme(resourceConfigsJSON);
+        List<String> schemes = getOtherSchemes();
+        boolean isDefaultAvailable = isDefaultGiven(resourceConfigsJSON);
+
+        for (String pathString : swagger.getPaths().keySet()) {
+            Path path = swagger.getPath(pathString);
+            Map<HttpMethod, io.swagger.models.Operation> operationMap = path.getOperationMap();
+            for (Map.Entry<HttpMethod, io.swagger.models.Operation> entry : operationMap.entrySet()) {
+                io.swagger.models.Operation operation = entry.getValue();
+                URITemplate template = new URITemplate();
+                template.setHTTPVerb(entry.getKey().name().toUpperCase());
+                template.setHttpVerbs(entry.getKey().name().toUpperCase());
+                template.setUriTemplate(pathString);
+                List<String> opScopesofDefault = getScopeOfOperations(SWAGGER_SECURITY_SCHEMA_KEY, operation);
+                if (opScopesofDefault == null) {
+                    opScopesofDefault = new ArrayList<>();
+                }
+                //Handling scopes in resources which do not belong to 'default' scheme
+                if (!isDefaultAvailable) {
+                    for (int i = 0; i < schemes.size(); i++) {
+                        List<String> opScopesOfOthers = getScopeOfOperations(schemes.get(i), operation);
+                        if (!opScopesOfOthers.isEmpty()) {
+                            if (opScopesOfOthers.size() == 1) {
+                                String firstScope = opScopesOfOthers.get(0);
+                                Scope scope = APIUtil.findScopeByKey(scopes, firstScope);
+                                if (scope == null) {
+                                    throw new APIManagementException("Scope '" + firstScope + "' not found.");
+                                }
+                                Scope duplication = scope;
+                                template.setScope(duplication);
+                                template.setScopes(duplication);
+                            } else {
+                                for (String scope : opScopesOfOthers) {
+                                    Scope scopeObj = new Scope();
+                                    scopeObj.setKey(SWAGGER_SECURITY_SCHEMA_KEY);
+                                    scopeObj.setName(SWAGGER_SECURITY_SCHEMA_KEY);
+                                    template.setScopes(scopeObj);
+                                }
+                            }
```

Comment:
What does the else part do?

Reply:
Yeah . But it is already implemented at getScopes() method also. So, I assumed it is implemented in case if we going to provide support in the future.